### PR TITLE
Bump AdMob SDK to 19.7.0 and replace deprecated APIs by new APIs

### PR DESCRIPTION
**Pull request by leonardlabat**

### CODE REVIEW

**Review by vguerci — DISMISSED** [review]
I guess you're doing the upgrade in two stages? v20 being the second?

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 1]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
       }
 
       @Override
-      public void onAdFailedToLoad(int i) {
+      public void onAdFailedToLoad(LoadAdError loadAdError) {
+        LoggerFactory.getLogger(DfpHeaderBiddingFunctionalTest.class)
+            .debug("onAdFailedToLoad for reason %d. "
+                    + "See: https://developers.google.com/android/reference/com/google/android/gms/ads/doubleclick/PublisherAdRequest",
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:697]: The URL can be removed. It was added as a pointer to the doc to explain the meaning of the `int` from old listener. Now that the listener use a proper `LoadAdError` enum, I guess it is no longer needed.

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 2]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
...
+      public void onAdLoaded(@NonNull AdManagerInterstitialAd adManagerInterstitialAd) {
+        _adManagerInterstitialAd = adManagerInterstitialAd;
+        isLoaded.countDown();
+      }
+
+      @Override
+      public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
         LoggerFactory.getLogger(DfpHeaderBiddingFunctionalTest.class)
             .debug("onAdFailedToLoad for reason %d. "
                     + "See: https://developers.google.com/android/reference/com/google/android/gms/ads/doubleclick/PublisherAdRequest",
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:720]: Same than above

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 3]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
...
+        LoggerFactory.getLogger(DfpHeaderBiddingFunctionalTest.class)
+            .debug("onAdFailedToLoad for reason %d. "
+                    + "See: https://developers.google.com/android/reference/com/google/android/gms/ads/doubleclick/PublisherAdRequest",
+                loadAdError);
+        isLoaded.countDown();
+      }
+    }
+
+    public class SyncAdManagerInterstitialAdLoadCallback extends AdManagerInterstitialAdLoadCallback {
+      private AdManagerInterstitialAd _adManagerInterstitialAd;
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:704]: Generally there is no Hungarian notation on Java/Android code to indicate the privacy of a variable. You can remove the _ in front of the name.

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 4]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
...
+      public void onAdFailedToLoad(LoadAdError loadAdError) {
+        LoggerFactory.getLogger(DfpHeaderBiddingFunctionalTest.class)
+            .debug("onAdFailedToLoad for reason %d. "
+                    + "See: https://developers.google.com/android/reference/com/google/android/gms/ads/doubleclick/PublisherAdRequest",
+                loadAdError);
+        isLoaded.countDown();
+      }
+    }
+
+    public class SyncAdManagerInterstitialAdLoadCallback extends AdManagerInterstitialAdLoadCallback {
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:703]: The visiblity of this class could be package private.

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 5]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
...
+      private AdManagerInterstitialAd _adManagerInterstitialAd;
+
+      public AdManagerInterstitialAd getAdManagerInterstitialAd() {
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:706]: Nit: in a simple callback in the test codebase, I wouldn't mind if you omit the getter :)

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 6]
```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
...
+        return _adManagerInterstitialAd;
+      }
+
+      @Override
+      public void onAdLoaded(@NonNull AdManagerInterstitialAd adManagerInterstitialAd) {
+        _adManagerInterstitialAd = adManagerInterstitialAd;
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:712]: If this side effect bother you, maybe could you rename the attribute to reflect what it is: `lastInterstitialAd` ?

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 7]
```diff
@@ -554,39 +563,46 @@ private String loadDfpHtmlInterstitial(InterstitialAdUnit adUnit) throws Excepti
...
-    PublisherAdRequest request = builder.build();
+    RequestConfiguration requestConfiguration = new RequestConfiguration.Builder().setTestDeviceIds(Collections.singletonList(AdRequest.DEVICE_ID_EMULATOR)).build();
+    MobileAds.setRequestConfiguration(requestConfiguration);
+
+    AdManagerAdRequest request = builder.build();
 
-    PublisherInterstitialAd publisherInterstitialAd = createDfpInterstitialView();
-    DfpSync dfpSync = new DfpSync(publisherInterstitialAd);
+    DfpSync dfpSync = new DfpSync();
+    SyncAdManagerInterstitialAdLoadCallback syncAdManagerInterstitialAdLoadCallback = dfpSync.getSyncAdManagerInterstitialAdLoadCallback();
```
Comment (TurpIF) [inline publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:572]: Nit: That's a really long name and the context is quite clear. What do you think about shorten it a little bit ?

**Review by TurpIF — CHANGES_REQUESTED** [review]
Could you also change the `com.google.android.gms.ads.doubleclick.PublisherAdRequest` in `DfpHeaderBiddingNoDfpTest` class (or append a new class) ? It is a precheck to assert that the `DfpHeaderBiddingNoDfpTest` tests are working without DFP in the classpath.

Do you also mind updating the `CHANGELOG.md` ?

**Review by TurpIF — CHANGES_REQUESTED** [review]

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 8]
```diff
@@ -648,16 +664,19 @@ private void givenLiveBiddingEnabled() {
     when(config.isLiveBiddingEnabled()).thenReturn(true);
   }
 
-  private static final class DfpSync {
+  public static final class DfpSync {
```
[the remark this answers is not in the document]
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:667]: Looks like that if `DfpSync` visibility is set to `private` the I cannot import `SyncAdManagerInterstitialAdLoadCallback`.
However, using the package private visibility seems sufficient enough.
Does it sound ok to you ?

[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java:667]: Ah yes, that's right. Sure ok for package private.

File: app/src/main/java/com/criteo/testapp/DfpActivity.java [hunk 9]
```diff
@@ -58,72 +65,74 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    loadAdView(adManagerAdView, BANNER);
   }
 
-  private void loadAdView(PublisherAdView publisherAdView, AdUnit adUnit) {
-    PublisherAdRequest.Builder builder = new PublisherAdRequest.Builder();
-    builder.addTestDevice(AdRequest.DEVICE_ID_EMULATOR);
+  private void loadAdView(AdManagerAdView adManagerAdView, AdUnit adUnit) {
+    AdManagerAdRequest.Builder builder = new AdManagerAdRequest.Builder();
+
+    RequestConfiguration requestConfiguration = new RequestConfiguration.Builder().setTestDeviceIds(Collections.singletonList(AdRequest.DEVICE_ID_EMULATOR)).build();
```
[the remark this answers is not in the document]
Comment (leonardlabat) [reply app/src/main/java/com/criteo/testapp/DfpActivity.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 7 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 6 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: Ah good suggestion ! done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 5 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 4 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 3 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: Ok, noted ! done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java (hunk 1 above)
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 10]
```diff
@@ -554,39 +563,46 @@ private String loadDfpHtmlInterstitial(InterstitialAdUnit adUnit) throws Excepti
...
+          syncAdManagerInterstitialAdLoadCallback
+      );
+    });
     dfpSync.waitForBid();
 
     View interstitialView = getRootView(webViewLookup.lookForResumedActivity(() -> {
-      runOnMainThreadAndWait(publisherInterstitialAd::show);
+      runOnMainThreadAndWait(() -> {
+        AdManagerInterstitialAd adManagerInterstitialAd = syncAdManagerInterstitialAdLoadCallback.getAdManagerInterstitialAd();
+        if (adManagerInterstitialAd != null) {
```
[the remark this answers is not in the document]
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 11]
```diff
@@ -554,39 +563,46 @@ private String loadDfpHtmlInterstitial(InterstitialAdUnit adUnit) throws Excepti
...
 
-    PublisherInterstitialAd publisherInterstitialAd = createDfpInterstitialView();
-    DfpSync dfpSync = new DfpSync(publisherInterstitialAd);
+    DfpSync dfpSync = new DfpSync();
+    SyncAdManagerInterstitialAdLoadCallback syncAdManagerInterstitialAdLoadCallback = dfpSync.getSyncAdManagerInterstitialAdLoadCallback();
 
-    runOnMainThreadAndWait(() -> publisherInterstitialAd.loadAd(request));
+    runOnMainThreadAndWait(() -> {
+      AdManagerInterstitialAd.load(
+          activityRule.getActivity().getApplicationContext(),
```
[the remark this answers is not in the document]
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

File: publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java [hunk 12]
```diff
@@ -528,16 +535,18 @@ private String loadDfpHtmlNative(NativeAdUnit adUnit) throws Exception {
...
       throws Exception {
     givenInitializedCriteo(adUnit);
 
     Builder builder = new Builder();
 
     loadBidAndWait(adUnit, builder);
 
-    builder.addTestDevice(AdRequest.DEVICE_ID_EMULATOR);
-    PublisherAdRequest request = builder.build();
+    RequestConfiguration requestConfiguration = new RequestConfiguration.Builder().setTestDeviceIds(Collections.singletonList(AdRequest.DEVICE_ID_EMULATOR)).build();
```
[the remark this answers is not in the document]
Comment (leonardlabat) [reply publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java]: done

**Review by TurpIF — APPROVED** [review]

**Pull request closed without merging**